src/test/java/com/commafeed/backend/feed/FeedParserTest.java

```diff
@@ -0,0 +1,207 @@
+        childOfParser parser = mock(childOfParser.class);
+        // Make a fake SyncEntry
+        SyndEntry fakeEntry = makeMockFakeSyndEntry();
+        Date now = new Date();
+        fakeEntry.setUpdatedDate(now);
+
+        // Get ealier date
+        Calendar calenEarly =  Calendar.getInstance();
+        calenEarly.set(928, 06,02);
+        Date testDateEarly = calenEarly.getTime();
+
+        when(parser.getEntryUpdateDate(fakeEntry)).thenReturn(now);
+        Assert.assertEquals(parser.getEntryUpdateDate(fakeEntry), now);
+
+        // Upon getting a date obj == null, suppose to get null (if publish date is null)
+        fakeEntry.setUpdatedDate(null);
+        when(parser.getEntryUpdateDate(fakeEntry)).thenReturn(null);
+        Assert.assertEquals(parser.getEntryUpdateDate(fakeEntry), null);
+
+        // Let's setup the publish date and expect the publish date even if UpdatedDate is null
+        fakeEntry.setPublishedDate(testDateEarly);
+        fakeEntry.setUpdatedDate(null);
+        when(parser.getEntryUpdateDate(fakeEntry)).thenReturn(testDateEarly);
+        Assert.assertEquals(parser.getEntryUpdateDate(fakeEntry), testDateEarly);
+    }
+    @Test
+    public void testGetContent(){
+        // In this section , it was not possible to use Mockito as the library org.apache.common.lang3
+        // was not initialized correctly using mock(class)...
+        childOfParser parser = new childOfParser();
+        // Make a fake SyncEntry
+        SyndEntry fakeEntryWithNothing = mock(SyndEntry.class);
+        // Test with no value
+        Assert.assertEquals(parser.getContent(fakeEntryWithNothing), null);
+
+        SyndEntry fakeEntry = makeFakeSyndEntry();
+        String expectedValue = fakeEntry.getContents().get(0).getValue();
+
+        Assert.assertEquals(parser.getContent(fakeEntry), expectedValue);
+    }
+    @Test
+    public void testGetTitle(){
+        // In this section , it was not possible to use Mockito as the library org.apache.common.lang3
+        // was not initialized correctly using mock(class)...
+        childOfParser parser = new childOfParser();
+        // Make a fake SyncEntry
+        SyndEntry fakeEntry = makeFakeSyndEntry();
+        String expectedTitle = fakeEntry.getTitle();
+        Assert.assertEquals(parser.getTitle(fakeEntry), expectedTitle);
+
+    }
+
+    /**
+     * This test will cover the main method: parse(...)
+     * Inside the class FeedParser...
+     */
+    @Test
+    public void testParse(){
+        childOfParser parser = new childOfParser();
+        FetchedFeed fetchedFeed = mock(FetchedFeed.class);
+        String fakeURL = "https://www.w3schools.com";
+        byte[] xml = makeFakeXMLFeed();
+        try{
+            fetchedFeed = parser.parse(fakeURL, xml);
+            Assert.assertEquals(fetchedFeed.getTitleForTest(), "W3Schools Home Page");
+        }catch (Exception ee){
+            ee.printStackTrace();
+        }
+    }
+
```

Comment:
For tests, I've heard that it is good practice not to make helper functions but instead to keep everything in the body of the test so that all you need to understand for the test is right before your eyes. It makes sense when you consider that each unit test should be totally independent from each other. Copy-paste is tolerable.

Reply:
aaaaah okay I will change it for next iteration... !!!